src/libs/SearchUtils.ts

```diff
@@ -5,12 +5,31 @@ import type {ReportListItemType, TransactionListItemType} from '@components/Sele
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import type * as OnyxTypes from '@src/types/onyx';
-import type {SearchDataTypes, SearchTypeToItemMap} from '@src/types/onyx/SearchResults';
+import type {SearchAccountDetails, SearchDataTypes, SearchTypeToItemMap, SectionsType} from '@src/types/onyx/SearchResults';
 import getTopmostCentralPaneRoute from './Navigation/getTopmostCentralPaneRoute';
 import navigationRef from './Navigation/navigationRef';
 import type {RootStackParamList, State} from './Navigation/types';
+import * as TransactionUtils from './TransactionUtils';
 import * as UserUtils from './UserUtils';
 
+type SortOrder = ValueOf<typeof CONST.SORT_ORDER>;
+type SearchColumnType = ValueOf<typeof CONST.SEARCH_TABLE_COLUMNS>;
+
+const columnNamesToSortingProperty = {
+    [CONST.SEARCH_TABLE_COLUMNS.TO]: 'formattedTo' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.FROM]: 'formattedFrom' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.DATE]: 'date' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.TAG]: 'tag' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.MERCHANT]: 'formattedMerchant' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.TOTAL]: 'formattedTotal' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.CATEGORY]: 'category' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.TYPE]: 'type' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.ACTION]: 'action' as const,
+    [CONST.SEARCH_TABLE_COLUMNS.DESCRIPTION]: null,
```

Comment:
The description can be used as sort column (https://github.com/Expensify/App/issues/44431)